flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -1099,14 +1100,18 @@ public CompletableFuture<Acknowledge> shutDownCluster(
     @Override
     public CompletableFuture<CoordinationResponse> deliverCoordinationRequestToCoordinator(
             JobID jobId,
-            OperatorID operatorId,
+            String userDefinedOperatorId,
             SerializedValue<CoordinationRequest> serializedRequest,
             Time timeout) {
         return performOperationOnJobMasterGateway(
                 jobId,
                 gateway ->
                         gateway.deliverCoordinationRequestToCoordinator(
-                                operatorId, serializedRequest, timeout));
+                                new OperatorID(
+                                        StreamGraphHasherV2.generateUserSpecifiedHash(
```

Comment:
Here it's hard to tell whether `StreamGraphHasherV2` should be used.
I think It's better to add a static method `OperatorID createOperatorId(bytes[] hash)` in `StreamJobGraphGenerator` because it explicitly uses `StreamGraphHasherV2`.